kiwi/utils/temporary.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) 2021 SUSE Linux GmbH.  All rights reserved.
+#
+# This file is part of kiwi.
+#
+# kiwi is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# kiwi is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with kiwi.  If not, see <http://www.gnu.org/licenses/>
+#
+from typing import IO
+
+from tempfile import (
+    NamedTemporaryFile,
+    TemporaryDirectory
+)
+
+# project
+import kiwi.defaults as defaults
+
+
+class Temporary:
+    """
+    **Provides namespace to handle temporary files and directories**
+    """
+    def __init__(
+        self, dir: str = defaults.TEMP_DIR, prefix: str = '',
+        delete: bool = True
```

Comment:
I am wondering if the delete attribute shouldn't be a parameter in `new_file` method rather part of the Temporary object. My reasoning is simple, this flag does not affect  `new_dir` method, so from an API perspective this might be confusing. Is there any place we are using it with `delete=false`? I hope not 😅 Probably we could omit this parameter and always force the cleanup.

Anyway this is a really minor detail, I can perfectly live with the current implementation :+1:

Reply:
Well, we could make it affect directories ourselves in the future, right?

Reply:
I think we don't want the temporary data to stay. Thus I agree with @davidcassany that it might be the best to no expose the delete feature at all. There is one place in the kiwi code that used it, in kiwi/system/setup.py create_recovery_archive() the created tmpfile was created with delete=False. It is done that way because the
generated tmpfile gets moved. If you move the file away prior the NamedTemporaryFile cleanup step it ends in
an exception

```python
import os
from tempfile import NamedTemporaryFile

x = NamedTemporaryFile()

print(x.name)

os.unlink(x.name)
```

I will refactor the code in kiwi as we only have one place where this issue exists

Reply:
@davidcassany pushed two commits to handle this better